### ISSUE TITLE
Supervisor and Caddy 2

### ISSUE THREAD

**Issue by johandalabacka** [lucaslorentz/caddy-run#14]
Hi. More of a question than an issue. How are your plans to support Caddy 2?

**Comment by mholt** [comment]
FWIW, I'd be interested in collaborating on an official (i.e. in the caddyserver organization) module for process supervision in Caddy 2. A lot of new and interesting possibilities here.

**Comment by lucaslorentz** [comment]
@mholt 
That's very nice.
I can update the readme page with links to the new module, let me know when you have a repo for it.

**Comment by Baldinof** [comment]
Hi!

Is there any plans on this?

I really enjoyed this plugin on caddy v1 :)

@lucaslorentz if you don't have time / plan for it I think I could find some time to work on this.

@mholt are you still OK to make this an official module?

**Comment by lucaslorentz** [comment]
@Baldinof 
I wasn't planning to work on it anytime soon.
But I can help you if you need any assistance.

**Comment by mholt** [comment]
> are you still OK to make this an official module?

Yeah, we can depending on its popularity/usefulness -- I don't have much spare bandwidth to collaborate right now though. (My earlier comment was from well over a year ago!) The only difference is where the code lives, any repo will do - there's no technical difference.

Would be cool to see one!

**Comment by DanyHenriquez** [comment]
@mholt I have read the documentation and think i can do this. I would make a PR this weekend with a simple implementation to monitor a process

**Comment by Baldinof** [comment]
Hi! 

I totally forgot to respond here, I was planning to make a PR too but I don't know when I will find the time to finish it.

For the logging part, I think the processes outputs could be written in a specific log field. This way it could be sent to stdout / stderr via something like this:

```
log {
  format single_field "supervisor_stdout"
}
```

**Comment by mholt** [comment]
(FWIW I intend to deprecate the `single_field` log encoder sooner or later.)

**Comment by DanyHenriquez** [comment]
So i see we want to use supervisor lib for this. Is there any library which has already been decided on?

**Comment by mholt** [comment]
Remind me to upload some pictures I took of the Windows Task Scheduler this last week. (I'm [waiting for GitHub to fix EXIF stripping](https://twitter.com/natfriedman/status/1389589163159154693) before I post them here.) I think it has a lot of useful customization options that could be good inspiration for this kind of module.
